pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -291,23 +288,67 @@ func (pl *PodTopologySpread) isSchedulableAfterNodeChange(logger klog.Logger, po
 		return framework.Queue, err
 	}
 
+	var originalNodeMatching, modifiedNodeMatching bool
+	if originalNode != nil {
+		originalNodeMatching = nodeLabelsMatchSpreadConstraints(originalNode.Labels, constraints)
+	}
+	if modifiedNode != nil {
+		modifiedNodeMatching = nodeLabelsMatchSpreadConstraints(modifiedNode.Labels, constraints)
+	}
+
+	// We return Queue in the following cases:
+	// 1. Node/UpdateNodeTaint:
+	//  - The modified node match the pod's topology spread constraints, and the original node and the modified node have different taints.
+	// 2. Node/UpdateNodeLabel:
+	//  - The original node matched the pod's topology spread constraints, but the modified node does not.
+	//  - The modified node matches the pod's topology spread constraints, but the original node does not.
+	//  - The modified node matches the pod's topology spread constraints, and the original node and the modified node have different label values.
+	// 3. Node/Add: The created node matches the pod's topology spread constraints.
+	// 4. Node/Delete: The original node matched the pod's topology spread constraints.
+
+	if originalNode != nil && modifiedNode != nil {
+		if modifiedNodeMatching && !equality.Semantic.DeepEqual(originalNode.Spec.Taints, modifiedNode.Spec.Taints) {
+			logger.V(5).Info("the node is updated and now has different taints, and the pod may be schedulable now",
+				"pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+			return framework.Queue, nil
+		}
+
+		if !originalNodeMatching && modifiedNodeMatching {
```

Comment:
```suggestion
		if originalNodeMatching != modifiedNodeMatching {
```
We can merge L316 and L322?